internal/services/compute/managed_disks_data_source.go

```diff
@@ -0,0 +1,277 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+)
+
+//
+// type ManagedDisksDataSource struct{}
+// type ManagedDisksDataSourceModel struct {
+// 	ResourceGroupName string `tfschema:"resource_group_name"`
+// 	Disks             []Disk `tfschema:"disk"`
+// }
+//
+// type Disk struct {
+// 	Name                string              `tfschema:"name"`
+// 	CreateOption        string              `tfschema:"create_option"`
+// 	DiskAccessId        string              `tfschema:"disk_access_id"`
+// 	DiskEncryptionSetId string              `tfschema:"disk_encryption_set_id"`
+// 	DiskIOPSReadWrite   int64               `tfschema:"disk_iops_read_write"`
+// 	DiskMBPSReadWrite   int64               `tfschema:"disk_mbps_read_write"`
+// 	DiskSizeMB          int64               `tfschema:"disk_size_gb"`
+// 	EncryptionSettings  []EncryptionSetting `tfschema:"encryption_settings"`
+// 	Location            string              `tfschema:"location"`
+// 	ImageReferenceID    string              `tfschema:"image_reference_id"`
+// 	NetworkAccessPolicy string              `tfschema:"network_access_policy"`
+// 	OSType              string              `tfschema:"os_type"`
+// 	SourceResourceID    string              `tfschema:"source_resource_id"`
+// 	SourceURI           string              `tfschema:"source_uri"`
+// 	StorageAccountID    string              `tfschema:"storage_account_id"`
+// 	StorageAccountType  string              `tfschema:"storage_account_type"`
+// 	Tags                map[string]string   `tfschema:"tags"`
+// 	Zones               []string            `tfschema:"zones"`
+// }
+//
+// type EncryptionSetting struct {
+// 	Enabled bool `tfschema:"enabled"`
+// 	DiskEncryptionKey []DiskEncryptionKey `tfschema:"disk_encryption_key"`
+// }
+
+func dataSourceManagedDisks() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceManagedDisksRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"resource_group_name": commonschema.ResourceGroupNameForDataSource(),
+
+			"disk": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"create_option": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"disk_access_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"disk_encryption_set_id": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"disk_iops_read_write": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+
+						"disk_mbps_read_write": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+
+						"disk_size_gb": {
```

Comment:
While I assume this is a copy-paste from the singular one, since it is a net new data source I think we should update it to follow our documented naming conventions

```suggestion
						"disk_size_in_gb": {
```

Reply:
I was torn on this one, and erred on consistent user UX and planned to catch them up in sync later. Given this call out however, I'm happy to swing the other way and I'll factor it in to the refactor to typed that I apparently lost the commits for in stashes clean-up 🙈